Microsoft.Toolkit.Sample.Wpf.App/Program.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using Windows.UI.Xaml.Markup;
+
+namespace Microsoft.Toolkit.Sample.Wpf.App
+{
+    public static class Program
+    {
+        enum StartupKind
+        {
+            MultiThread,
+            CustomAppSettings,
+            Normal,
+        };
+
+        /// <summary>
+        /// The main entry point for the application.
+        /// </summary>
+        [STAThread]
+        public static void Main()
+        {
+            var startupKind = StartupKind.Normal;
+
+            if (startupKind == StartupKind.CustomAppSettings)
+            {
+                using (var xamlApp = new Win32.UI.XamlHost.XamlApplication()
+                {
+                    Resources = new Windows.UI.Xaml.ResourceDictionary()
+                        {
+                            { "MyResourceKey", "MyValue" },
+                        },
+                    MetadataProviders = {
+                        new MyMetadataProvider(),
+                    },
+                })
```

Comment:
Here MyMetadataProvider gets added too late. The .ctor already uses MetadataProviders

Reply:
Resolved now, you should see only one instance.

---
In reply to: [259245254](https://github.com/windows-toolkit/Microsoft.Toolkit.Win32/pull/82#discussion_r259245254) [](ancestors = 259245254)